test/e2e/external_oidc_test.go

```diff
@@ -2,54 +2,244 @@ package e2e
 
 import (
 	"context"
+	"crypto/rsa"
 	"crypto/tls"
+	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"io"
+	"math/big"
 	"net/http"
-	"net/url"
-	"os"
+	"strings"
 	"testing"
 	"time"
 
 	configv1 "github.com/openshift/api/config/v1"
 	features "github.com/openshift/api/features"
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned"
 	operatorversionedclient "github.com/openshift/client-go/operator/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
 	test "github.com/openshift/cluster-authentication-operator/test/library"
 	"github.com/stretchr/testify/require"
+	apiregistrationclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
 
 	authenticationv1 "k8s.io/api/authentication/v1"
-	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+
+	"github.com/golang-jwt/jwt/v5"
 )
 
 const (
-	oidcClientId      = "admin-cli"
-	oidcAudience      = "openshift-aud"
-	oidcGroupsClaim   = "groups"
-	oidcUsernameClaim = "email"
-
-	kasNamespace = "openshift-kube-apiserver"
+	oidcClientId     = "admin-cli"
+	oidcGroupsClaim  = "groups"
+	oidcGroupsPrefix = ""
 
-	// set if a specific CA bundle is needed for the OIDC provider
-	oidcCABundleConfigMap          = ""
-	oidcCABundleConfigMapNamespace = ""
+	managedNS = "openshift-config-managed"
+	authCM    = "auth-config"
 )
 
-type testClient struct {
-	t *testing.T
+func TestExternalOIDCWithKeycloak(t *testing.T) {
+	testCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	tc, err := newTestClient(t)
```

Comment:
Nit: More descriptive name like `testClient`? A few times going through this PR I got confused because I commonly see `tc` represented as "test case" so it threw me off a bit that this meant "test client".